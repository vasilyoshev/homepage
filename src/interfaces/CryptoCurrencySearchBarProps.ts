export interface CryptoCurrencySearchBarProps {
    type: string;
    placeholder: string;
    onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
}
