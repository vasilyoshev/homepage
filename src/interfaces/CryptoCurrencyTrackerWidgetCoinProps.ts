export interface CoinProps {
    name: string;
    image: string;
    symbol: string;
    marketcap: number;
    price: number;
    priceChange: number;
}
