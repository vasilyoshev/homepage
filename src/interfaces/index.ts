export * from './LoadingSliceState';
export * from './TodoWidgetProps';
export * from './WidgetsSliceState';
export * from './WidgetProps';
export * from './CryptoCurrencyTrackerWidgetResponse';
export * from './CryptoCurrencyTrackerWidgetInfoProps';
export * from './CryptoCurrencyTrackerWidgetSearchBarProps';
export * from './TimerWidgetControlsProps';
export * from './WeatherWidgetLocationSearchProps';
export * from './WeatherWidgetLocationTableProps';
export * from './WeatherWidgetModel';
