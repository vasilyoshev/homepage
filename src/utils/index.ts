export * from './handleLoadingState';
export * from './testUtils';
