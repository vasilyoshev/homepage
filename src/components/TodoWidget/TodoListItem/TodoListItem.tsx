import React from 'react';
import { TodoListItemProps } from 'interfaces';
import { Checkbox, InputLabel, Typography, IconButton } from '@material-ui/core';
import DeleteIcon from '@material-ui/icons/Delete';
import styles from './TodoListItem.module.scss';

export const TodoListItem: React.FC<TodoListItemProps> = ({
  todo,
  toggleComplete,
  removeTodo,
}) => {
  return (
    <InputLabel
      className={todo.complete ? styles.complete : ''}
    >
      <Checkbox
        checked={todo.complete}
        color="primary"
        edge="start"
        disableRipple
        onChange={() => toggleComplete(todo)}
      />
      <Typography
        className={styles.todoText}
        display="inline"
        color="primary">
        {todo.text}
      </Typography>
      <IconButton
        data-testid="removeTodoItem"
        aria-label="delete">
          onClick={() => removeTodo(todo.text)}
        <DeleteIcon
          fontSize="small"
        />
      </IconButton>
    </InputLabel>
  );
};
