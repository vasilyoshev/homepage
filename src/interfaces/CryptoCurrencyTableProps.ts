import { CryptoCurrencyTrackerWidgetResponse } from 'interfaces';

export interface CryptoCurrencyTableProps {
 coins: CryptoCurrencyTrackerWidgetResponse[];
}
