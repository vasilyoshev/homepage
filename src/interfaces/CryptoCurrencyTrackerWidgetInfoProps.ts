import { CryptoCurrencyTrackerWidgetResponse } from 'interfaces';

export interface CryptoCurrencyTrackerWidgetInfoProps {
    coins: CryptoCurrencyTrackerWidgetResponse[];
}
