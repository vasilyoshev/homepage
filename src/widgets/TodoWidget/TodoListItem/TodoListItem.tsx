import { TodoListItemProps } from 'interfaces';
import React from 'react';
import todoListItem  from 'stylesheets/widgets/TodoListItem.module.scss';
export const TodoListItem: React.FC<TodoListItemProps> = ({
  todo,
  toggleComplete,
}) => {
  return (
    <li>
      <label className={todo.complete ? todoListItem.complete : ''}>
        <input
          type="checkbox"
          onChange={() => toggleComplete(todo)}
          checked={todo.complete}
        />
        {todo.text}
      </label>
    </li>
  );
};
