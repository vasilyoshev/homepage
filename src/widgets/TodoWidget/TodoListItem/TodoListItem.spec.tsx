import { Todo } from 'interfaces';
import { render } from 'utils';
import { TodoListItem } from './TodoListItem';

describe('TodoListItem', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<TodoListItem toggleComplete={jest.fn()} todo={{} as Todo} />);
    expect(asFragment()).toMatchSnapshot();
  });
}); 
  

