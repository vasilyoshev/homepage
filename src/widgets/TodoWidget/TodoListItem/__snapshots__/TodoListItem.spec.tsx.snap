// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem renders successfully 1`] = `
<DocumentFragment>
  <li>
    <label
      class=""
    >
      <input
        type="checkbox"
      />
    </label>
  </li>
</DocumentFragment>
`;
